Feature watch task

kaiwollhaupt opened the pull request:
Merge branch feature watch-task into 2.x-dev .

cherryann committed:
[+TASK] add new css linter stylelint, todo: configuration

cherryann committed:
[+TASK] add new css linter stylelint - add configuration

cherryann committed:
[+TASK] remove scsslint, transfer/add config rules from scsslint to stylelint, adjust package.json

cherryann committed:
[+TASK] remove scsslint, transfer/add config rules from scsslint to stylelint, adjust package.json

cherryann committed:
add stylelint to package.json, remove sass-lint config, clean files

TatjanaRo committed:
update stylelint implementation to recommended way, includes stylelint-config-standard

TatjanaRo committed:
include stylelint-config-standard

Commit:
Add sourcemaps in postcss autoprefixer, change livereload, rename scss partials

Commit:
Merge branch 'feature/add-stylelint' into feature/watch-task

Delete grunt-stylelint, add stylelint for grunt-postcss, comment css rules, change watch task

Commit:
Change .stylelintrc and .stylelintrcComments

Commit:
Change watch, postcss grunt tasks

TatjanaRo committed:
ignore vendor files in postcss config, fixes #43

Commit:
Remove postcss-stylelint plugin and add grunt-stylelint to use checkstyle report for ci and .stylelintignore file, resolves #44 #43
Update grunt tasks and package.json. Move deprecated lintingComments files to docs folder.
Delete stylelint disable/enable comments in scss partials

mariusbuescher commented on grunt/postcss.js:
```diff
@@ -1,18 +1,39 @@
-                    } )
+                    } ),
+                    require( 'postcss-inline-svg' ),
+                    require( 'postcss-svgo' )
```
This is new. `postcss-inline-svg` and `postcss-svgo` are not yet part of the standard set up. Please also remove those plugins from the `package.json`

kaiwollhaupt replied:
Plugins removed.

mariusbuescher commented on grunt/requirejs.js:
```diff
@@ -13,7 +13,7 @@ module.exports = function( grunt, options ) {
                 dir: '<%= distPath %>js',
 
-                fileExclusionRegExp: /\.(njs|scss|gitkeep)/,
+                fileExclusionRegExp: /\.(njs|scss|gitkeep|DS_Store)/,
```
Do we really need this?

kaiwollhaupt replied:
Plugin grunt-contrib-requirejs threw an exception because it took this file into account. Since i can't reproduce this, i undo this change.

mariusbuescher commented on grunt/scsslint.js:
```diff
@@ -1,20 +0,0 @@
-        compact: true,
-        maxBuffer: NaN
-    }
-};
```
Removing `scss-lint` would be a breaking change. We are developing for 2.x here, so no breaking changes. Please revert the removal of `scss-lint`.

kaiwollhaupt replied:
By merging branch feature/add-stylelint to feature/watch-task plugins grunt-contrib-csslint and grunt-scss-lint were removed. Since this is a breaking change I took both plugins in again.

mariusbuescher commented on src/components/sass/vendor/_normalize.scss:
```diff
@@ -0,0 +1,447 @@
+
+[hidden] {
+    display: none;
+}
```
Why did you move those files? Is a leading underscore really neccesary?

kaiwollhaupt replied:
This is a naming convention for partials for which no own CSS file should be created.
http://sass-lang.com/documentation/file.SASS_REFERENCE.html#Partials__partials

mariusbuescher replied:
Thats ok with me.

mariusbuescher commented on docs/deprecated_scss-linting/.stylelintrcComments:
```diff
@@ -0,0 +1,227 @@
+
+    "string-quotes": "single",                                  // stylistic issue                                                              -> stylelintscssrc
+  }
+}
```
This file is of ne need, please remove it.

kaiwollhaupt replied:
File removed.

mariusbuescher commented on docs/deprecated_scss-linting/.stylelintscssrc:
```diff
@@ -0,0 +1,194 @@
+    "scss/dollar-variable-pattern": "^[a-z\\-_0-9]+$"
+  }
+}
+
```
Same here: not needed anymore, please remove

kaiwollhaupt replied:
File removed.

mariusbuescher commented on grunt/aliases.yaml:
```diff
@@ -2,9 +2,9 @@ development:
-        - 'postcss:autoprefixer'
         - 'lint'
+        - 'sass:development'
+        - 'postcss:cssDevelopment'
```
Maybe we can combine the `sass` and the `postcss` task to one `styles` task and call it combined, so we do not have to call the whole `development` task when we just want to build the css.

kaiwollhaupt replied:
Let's call this task stylesDevelopment to recognize if it's building the development or the production CSS.

mariusbuescher replied:
Please add one task for production as well.

mariusbuescher reviewed (changes requested):
So I just reviewed the changes you made. Seems to be more than needed and some things really introduce breaking changes.

Commit:
Delete .stylelintrcComments and .stylelintscssrc files,
add stylesDevelopment task in aliases.yaml,
remove postcss-inline-svg and postcss-svgo plugins,
revert grunt-scss-lint and grunt-contrib-csslint removal,
revert change in property fileExclusionRegExp of requirejs task

mariusbuescher commented on grunt/aliases.yaml:
```diff
@@ -27,6 +27,17 @@ prod:
+    description: 'Shortcut for stylesDevelopment task'
+    tasks:
+        - 'stylesDevelopment'
+
```
Please add one task for production as well.

Commit:
Add task stylesProduction in aliases.yaml
